### ISSUE TITLE
[Config Support]: Allow required_zones to specify object type

### ISSUE THREAD

**Issue by mike-lloyd03** [blakeblackshear/frigate#3472]
### Describe the problem you are having

I am trying to follow the guide for Ignoring Stationary Objects on my driveway. I have a zone where my cars are parked and I have a zone on the driveway where they drive. The driveway zone is where I have my notifications set to trigger. I would like the parking area to not trigger event recordings for stationary vehicles. So I'd like to use the `required_zones` option to only include when cars enter the driveway zone but also when people enter either zone. How can I configure this?

I'd expect the configuration might look something like this:

```yaml
cameras:
  cam1:
    record:
      events:
        required_zones:
          - name: driveway
            objects:
              - person
              - car
          - name: parking_area
            objects:
              - person
```

### Version

0.10.1-83481af

### Frigate config file

```yaml
detectors:
  coral:
    type: edgetpu
    device: usb

mqtt:
  host: 192.168.0.100
  port: 1883
  user: {FRIGATE_MQTT_USER}
  password: {FRIGATE_MQTT_PASSWORD}

cameras:
  Driveway:
    ffmpeg:
      inputs:
        # - path: rtmp://192.168.0.149/bcs/channel0_sub.bcs?channel=0&stream=0&user={FRIGATE_CAM_USER}&password={FRIGATE_CAM_PASSWORD}
        - path: http://192.168.0.149/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user={FRIGATE_CAM_USER}&password={FRIGATE_CAM_PASSWORD}
          roles:
            - detect
        # - path: rtmp://192.168.0.149/bcs/channel0_main.bcs?channel=0&stream=0&user={FRIGATE_CAM_USER}&password={FRIGATE_CAM_PASSWORD}
        - path: http://192.168.0.149/flv?port=1935&app=bcs&stream=channel0_main.bcs&user={FRIGATE_CAM_USER}&password={FRIGATE_CAM_PASSWORD}
          roles:
            - rtmp
            - record
    detect:
      width: 640
      height: 480
      fps: 7
    objects:
      track:
        - person
        - car
    motion:
      mask:
        - 0,0,640,0,640,114,574,107,491,73,309,70,0,141
    zones:
      parking_area:
        coordinates: 345,480,0,480,0,174,270,113,335,143,268,177,547,302
      driveway:
        coordinates: 289,170,490,238,614,98,546,82
    record:
      events:
        required_zones:
          - driveway

  FrontDoor:
    ffmpeg:
      inputs:
        # - path: rtmp://192.168.0.129/bcs/channel0_sub.bcs?channel=0&stream=0&user={FRIGATE_CAM_USER}&password={FRIGATE_CAM_PASSWORD}
        - path: http://192.168.0.129/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user={FRIGATE_CAM_USER}&password={FRIGATE_CAM_PASSWORD}
          roles:
            - detect
        # - path: rtmp://192.168.0.129/bcs/channel0_main.bcs?channel=0&stream=0&user={FRIGATE_CAM_USER}&password={FRIGATE_CAM_PASSWORD}
        - path: http://192.168.0.129/flv?port=1935&app=bcs&stream=channel0_main.bcs&user={FRIGATE_CAM_USER}&password={FRIGATE_CAM_PASSWORD}
          roles:
            - rtmp
            - record
    detect:
      width: 640
      height: 480
      fps: 7
    objects:
      track:
        - person
    motion:
      mask:
        - 640,0,640,179,534,171,372,123,232,82,184,0
      
  Backyard:
    ffmpeg:
      inputs:
        # - path: rtmp://192.168.0.102/bcs/channel0_sub.bcs?channel=0&stream=0&user={FRIGATE_CAM_USER}&password={FRIGATE_CAM_PASSWORD}
        - path: http://192.168.0.102/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user={FRIGATE_CAM_USER}&password={FRIGATE_CAM_PASSWORD}
          roles:
            - detect
        # - path: rtmp://192.168.0.102/bcs/channel0_main.bcs?channel=0&stream=0&user={FRIGATE_CAM_USER}&password={FRIGATE_CAM_PASSWORD}
        - path: http://192.168.0.102/flv?port=1935&app=bcs&stream=channel0_main.bcs&user={FRIGATE_CAM_USER}&password={FRIGATE_CAM_PASSWORD}
          roles:
            - rtmp
            - record
    detect:
      width: 640
      height: 480
      fps: 7
    objects:
      track:
        - person
        - car
      filters:
        car:
          mask:
            - 0,480,640,480,640,55,551,53,422,89,204,90,0,156
        person:
          mask:
            - 640,0,0,0,0,165,222,83,640,108

  KidsCam:
    ffmpeg:
      inputs:
        - path: rtsp://{FRIGATE_CAM_USER}:{FRIGATE_CAM_PASSWORD}@192.168.0.135:554/cam/realmonitor?channel=1&subtype=1
          roles:
            - detect
        - path: rtsp://{FRIGATE_CAM_USER}:{FRIGATE_CAM_PASSWORD}@192.168.0.135:554/cam/realmonitor?channel=1&subtype=0
          roles:
            - rtmp
      input_args: -avoid_negative_ts make_zero -fflags nobuffer -flags low_delay -strict experimental -fflags +genpts+discardcorrupt -use_wallclock_as_timestamps 1
    detect:
      width: 640
      height: 480
      fps: 5
    record:
      enabled: false

ffmpeg:
  # hwaccel_args:
  #   - -hwaccel
  #   - vaapi
  #   - -hwaccel_device
  #   - /dev/dri/renderD128
  #   - -hwaccel_output_format
  #   - yuv420p
  input_args:
    - -avoid_negative_ts
    - make_zero
    - -fflags
    - nobuffer+genpts+discardcorrupt
    - -flags
    - low_delay
    - -strict
    - experimental
    - -analyzeduration
    - 1000M
    - -probesize
    - 1000M
    - -rw_timeout
    - "5000000"
  output_args:
    record:
      - -f
      - segment
      - -segment_time
      - 10
      - -segment_format
      - mp4
      - -reset_timestamps
      - 1
      - -strftime
      - 1
      - -c:v
      - copy
      - -c:a
      - aac

record:
  enabled: true
  retain:
    days: 10
    mode: all
  events:
    pre_capture: 3 # seconds
    post_capture: 5 # seconds
    retain:
      default: 10 # days
```

### Relevant log output

```shell
N/A
```

### Operating system

Other Linux

### Install method

Docker Compose

### Coral version

USB

**Comment by NickM-27** [comment]
Please check the [full config docs](https://docs.frigate.video/configuration/index)

```yaml

cameras:
  back:
    ffmpeg:
      inputs: ...
    record:
      events:
        required_zones:
          -front_steps
    zones:
      front_steps:
        coordinates: 545,1077,747,939,788,805
        objects:
          - person
```

**Comment by NickM-27** [comment]
What do you mean by "not detected"? Masks and zones work entirely differently

1. I think you have this but to be clear car needs to be in the list of tracked objects for that camera
2. How is your zone defined? Are you aware that an objects presence in a zone is calculated based in the bottom center of the bounding box?
3. A screenshot of the debug camera view with bounding boxes and zones enabled will be helpful

**Comment by NickM-27** [comment]
Scratch that I think I understand what you're asking now. You want to use zones so it's limited to having events only when an object enters an area of interest. 

A mask means the car is fully ignored / not tracked which is not optimal for a couple reasons. 
1. You want to know where the car is at all times. Even if it's just driving by the street you want to have the car tracked so if it enters a zone of interest the entire time the car was in view is recorded instead of only after it is outside the mask
2. Masks can lead to other unexpected behavior

As the docs state, object masks are only designed to be used for false positives

**Comment by mike-lloyd03** [comment]
Sorry, I mean cars in the `parking_area` zone are not triggering events which is what I want. 

![Screenshot_20220715-075318_1.png](https://user-images.githubusercontent.com/49411532/179249148-963e461a-c898-401f-b7d5-9d47714c976f.png)

And anything that happens outside these zones will not trigger an event, correct?

**Comment by NickM-27** [comment]
> And anything that happens outside these zones will not trigger an event, correct?

You will need to add `required_zones` under snapshots as well for this to work that way

**Comment by mike-lloyd03** [comment]
Okay I think that gets me where I want to be. Thank you for the help. I'll close this now and reopen if I run into another issue.
